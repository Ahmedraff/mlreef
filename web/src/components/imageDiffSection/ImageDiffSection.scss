.t-bold {
  font-weight: bold !important;
}

.image-diff-section {
  & .image-display {
    overflow-y: hidden;

    &.collapsed {
      height: 1px;
    }
  }

  & .image-dimensions {
    display: flex;
    justify-content: space-around;
    // margin-top: 0.5rem;
  }

  .solid-border {
    border: 1px solid;
  }

  .image-dimensions {
    text-align: center;
  }

  .basic-information-image {
    flex: 1;
    padding: 1em;
  }

  #image-div-section-container {
    margin-top: 2em;
  }

  .image-container {
    margin-right: 1em;
  }
}

.commit-per-date{
  border: 1px solid #adadad;
  border-radius: 0.3vw;
  margin-bottom: 8px;
}

.commit-per-date:last-child{
  border-bottom: 1px solid #adadad;
  margin-bottom: 2em;
}

.commit-header{
  margin-top: 0;
  border-bottom: 1px solid #e5e5e5;
  background-color: #e5e5e5;
  display: flex;
  border-radius: 0.3vw 0.3vw 0 0;

  p {
    margin-left: 1em;
    color: #1c2b40;
    font-size: 14px;
    font-weight: 600;
  }
}