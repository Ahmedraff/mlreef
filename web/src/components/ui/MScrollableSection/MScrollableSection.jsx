import React, { useEffect, useRef } from 'react';
import PropTypes, { any } from 'prop-types';

const MScrollableSection = ({
  children, handleOnScrollDown, className,
}) => {
  const containerRef = useRef();

  useEffect(() => {
    const onScrollCallback = () => {
      const lastEle = containerRef.current?.lastChild;
      const lastEleOffSet = lastEle?.offsetTop + lastEle?.clientHeight;
      const pageOffset = window.pageYOffset + window.innerHeight;
      const bottomOffset = 200;
      if (pageOffset > lastEleOffSet - bottomOffset) handleOnScrollDown();
    };
    window.addEventListener('scroll', onScrollCallback);

    return () => window.removeEventListener('scroll', onScrollCallback);
  }, []);

  return (
    <div ref={containerRef} className={className}>
      {children}
    </div>
  );
};

MScrollableSection.propTypes = {
  children: any,
  handleOnScrollDown: PropTypes.func.isRequired,
  className: PropTypes.string,
};

MScrollableSection.defaultProps = {
  className: '',
};

export default MScrollableSection;
