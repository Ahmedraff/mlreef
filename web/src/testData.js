export const imagesToRender = [
  {
    commitId: 'd80ab8ff15fb1f327bac0f5679b454cbec247759',
    parentCommitId: '6819ee9d2e8f7796887cc87af2d396c9894a2ebf',
    previousImage: {},
    nextImage: {},
    fileName: 'color_0_0054_A.png',
  },
];

export const mockMatchDataCommitDet = {
  path: '/my-projects/:projectId/commit/:commitId',
  url: '/my-projects/12395599/commit/aede036f526ea5d591f5ee56c131ed8472483912',
  isExact: true,
  params: {
    projectId: '12395599',
    branch: 'master',
    file: '.gitignore',
    commitId: 'aede036f526ea5d591f5ee56c131ed8472483912',
  },
};

export const commitMockObject = {
  id: 'd80ab8ff15fb1f327bac0f5679b454cbec247759',
  short_id: 'd80ab8ff',
  created_at: '2019-11-09T14:24:40.000Z',
  parent_ids: [
    '6819ee9d2e8f7796887cc87af2d396c9894a2ebf',
  ],
  title: 'Replace color_0_0054_A.png',
  message: 'Replace color_0_0054_A.png',
  author_name: 'cpmlreef',
  author_email: 'cp@mlreef.com',
  authored_date: '2019-11-09T14:24:40.000Z',
  committer_name: 'cpmlreef',
  committer_email: 'cp@mlreef.com',
  committed_date: '2019-11-09T14:24:40.000Z',
  stats: {
    additions: 0,
    deletions: 0,
    total: 0,
  },
  status: null,
  last_pipeline: null,
  project_id: 14448940,
};

export const mockDiffArray = [
  {
    old_path: 'showcase/color_0_0054_A.png',
    new_path: 'showcase/color_0_0054_A.png',
    a_mode: '100644',
    b_mode: '100644',
    new_file: false,
    renamed_file: false,
    deleted_file: false,
    diff: 'Binary files a/showcase/color_0_0054_A.png and b/showcase/color_0_0054_A.png differ\n',
  },
];

export const projectsArrayMock = {
  projects: {
    all: [
      {
        id: 14448940,
        description: 'Categorical Classification of alphabets in the American Sign Language',
        name: 'Sign Language Classifier',
        name_with_namespace: 'MLReef / Sign Language Classifier',
        path: 'sign-language-classifier',
        path_with_namespace: 'mlreef/sign-language-classifier',
        created_at: '2019-09-23T08:51:23.983Z',
        default_branch: 'master',
        tag_list: [],
        ssh_url_to_repo: 'git@gitlab.com:mlreef/sign-language-classifier.git',
        http_url_to_repo: 'https://gitlab.com/mlreef/sign-language-classifier.git',
        web_url: 'https://gitlab.com/mlreef/sign-language-classifier',
        readme_url: 'https://gitlab.com/mlreef/sign-language-classifier/blob/master/README.md',
        avatar_url: 'https://gitlab.com/uploads/-/system/project/avatar/14448940/proxy.duckduckgo.png',
        star_count: 0,
        forks_count: 0,
        last_activity_at: '2019-11-10T15:14:17.804Z',
        namespace: {
          id: 5351747,
          name: 'MLReef',
          path: 'mlreef',
          kind: 'group',
          full_path: 'mlreef',
          parent_id: null,
          avatar_url: '/uploads/-/system/group/avatar/5351747/MLReef_Logo_POS_H_icon-01.png',
          web_url: 'https://gitlab.com/groups/mlreef',
        },
      },
    ],
    selectedProject: {
      id: 12395599,
      description: 'This is MLReefÂ´s demo repository.',
      name: 'demo',
      name_with_namespace: 'MLReef / demo',
      path: 'mlreef-demo',
      path_with_namespace: 'mlreef/mlreef-demo',
      created_at: '2019-05-18T09:34:29.346Z',
      default_branch: 'master',
      tag_list: [],
      ssh_url_to_repo: 'git@gitlab.com:mlreef/mlreef-demo.git',
      http_url_to_repo: 'https://gitlab.com/mlreef/mlreef-demo.git',
      web_url: 'https://gitlab.com/mlreef/mlreef-demo',
      readme_url: 'https://gitlab.com/mlreef/mlreef-demo/blob/master/README.md',
      avatar_url: 'https://gitlab.com/uploads/-/system/project/avatar/12395599/shutterstock_129655604-Converted-copy-text.png',
      star_count: 2,
      forks_count: 1,
      last_activity_at: '2019-11-11T10:36:06.189Z',
      namespace: {
        id: 5351747,
        name: 'MLReef',
        path: 'mlreef',
        kind: 'group',
        full_path: 'mlreef',
        parent_id: null,
        avatar_url: '/uploads/-/system/group/avatar/5351747/MLReef_Logo_POS_H_icon-01.png',
        web_url: 'https://gitlab.com/groups/mlreef',
      },
    },
  },
  users: [
    {
      id: 3839940,
      name: 'cpmlreef',
      username: 'cpmlreef',
      state: 'active',
      avatar_url: 'https://assets.gitlab-static.net/uploads/-/system/user/avatar/3839940/avatar.png',
      web_url: 'https://gitlab.com/cpmlreef',
    },
  ],
};


export const branchesMock = [
  {
    name: 'data-instance/664e1c60',
    commit: {
      id: '12e05078d142b2bee947e86d6a52db4a5f85eb65',
      short_id: '12e05078',
      created_at: '2019-11-01T21:08:23.000+00:00',
      parent_ids: null,
      title: 'Add pipeline results [skip ci]',
      message: 'Add pipeline results [skip ci]',
      author_name: 'mlreefdemo',
      author_email: 'rainer+mlreefdemo@systemkern.com',
      authored_date: '2019-11-01T21:08:23.000+00:00',
      committer_name: 'mlreefdemo',
      committer_email: 'rainer+mlreefdemo@systemkern.com',
      committed_date: '2019-11-01T21:08:23.000+00:00',
    },
    merged: false,
    protected: false,
    developers_can_push: false,
    developers_can_merge: false,
    can_push: true,
    default: false,
  },
  {
    name: 'data-instance/9ae9c8c0',
    commit: {
      id: '3e36bfab31a1d43457e00f46af9e6d31d4f55a66',
      short_id: '3e36bfab',
      created_at: '2019-11-14T20:32:52.000+00:00',
      parent_ids: null,
      title: 'Add pipeline results [skip ci]',
      message: 'Add pipeline results [skip ci]',
      author_name: 'mlreefdemo',
      author_email: 'rainer+mlreefdemo@systemkern.com',
      authored_date: '2019-11-14T20:32:52.000+00:00',
      committer_name: 'mlreefdemo',
      committer_email: 'rainer+mlreefdemo@systemkern.com',
      committed_date: '2019-11-14T20:32:52.000+00:00',
    },
    merged: false,
    protected: false,
    developers_can_push: false,
    developers_can_merge: false,
    can_push: true,
    default: false,
  },
  {
    name: 'master',
    commit: {
      id: '3e36bfab31a1d43457e00f46af9e6d31d4f55a66',
      short_id: '3e36bfab',
      created_at: '2019-11-14T20:32:52.000+00:00',
      parent_ids: null,
      title: 'Add pipeline results [skip ci]',
      message: 'Add pipeline results [skip ci]',
      author_name: 'mlreefdemo',
      author_email: 'rainer+mlreefdemo@systemkern.com',
      authored_date: '2019-11-14T20:32:52.000+00:00',
      committer_name: 'mlreefdemo',
      committer_email: 'rainer+mlreefdemo@systemkern.com',
      committed_date: '2019-11-14T20:32:52.000+00:00',
    },
    merged: false,
    protected: true,
    developers_can_push: false,
    developers_can_merge: false,
    can_push: true,
    default: true,
  },
];
