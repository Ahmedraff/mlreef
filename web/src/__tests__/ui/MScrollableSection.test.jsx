import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { filesInScrollableSection } from 'testData';
import ImageDiffSection from 'components/imageDiffSection/imageDiffSection';
import MScrollableSection from 'components/ui/MScrollableSection/MScrollableSection';

const className = 'diff-sections';
const handleOnScrollDown = jest.fn();
const children = filesInScrollableSection.map((file) => (
  <ImageDiffSection
    key={file.fileName}
    fileInfo={file.props.fileInfo}
    fileSize={file.props.fileSize}
    original={file.props.original}
    modified={file.props.modified}
  />
));

const setup = () => {
  const testWrapper = mount(
    <Router>
      <MScrollableSection
        className={className}
        handleOnScrollDown={handleOnScrollDown}
      >
        {children}
      </MScrollableSection>
    </Router>,
  );
  return testWrapper;
};

test('assert that snapshot matches', () => {
  const snapShot = renderer.create(
    <Router>
      <MScrollableSection
        className={className}
        handleOnScrollDown={handleOnScrollDown}
      >
        {children}
      </MScrollableSection>
    </Router>,
  ).toJSON();
  expect(snapShot).toMatchSnapshot();
});

describe('check scrollable section renders elements', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('contains children elements', () => {
    expect(wrapper.find('.diff-sections').children()).toHaveLength(3);
  });
});
