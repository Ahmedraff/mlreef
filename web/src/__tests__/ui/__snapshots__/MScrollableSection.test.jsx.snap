// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assert that snapshot matches 1`] = `
<div
  className="diff-sections"
>
  <div
    className="image-diff-section"
    id="image-div-section-container"
  >
    <div
      className="commit-per-date"
    >
      <div
        className="pipeline-modify-details"
      >
        <div
          className="basic-information-image"
        >
          <span
            id="image-modified-name"
          >
            5.jpg
          </span>
          <span>
            -
            5
            Kb
          </span>
        </div>
        <div
          className="filechange-info"
        >
          <button
            className="btn btn-sm btn-basic-dark px-3 mr-2"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fa fa-chevron-up"
            />
          </button>
          <button
            className="btn btn-sm btn-basic-dark mr-2"
            type="button"
          >
            Copy Path
          </button>
          <a
            className="btn btn-sm btn-basic-dark"
            href="/#foo"
            onClick={[Function]}
          >
            View File
          </a>
        </div>
      </div>
      <div
        className="image-display "
      >
        
        
      </div>
    </div>
  </div>
  <div
    className="image-diff-section"
    id="image-div-section-container"
  >
    <div
      className="commit-per-date"
    >
      <div
        className="pipeline-modify-details"
      >
        <div
          className="basic-information-image"
        >
          <span
            id="image-modified-name"
          >
            12.jpg
          </span>
          <span>
            -
            5
            Kb
          </span>
        </div>
        <div
          className="filechange-info"
        >
          <button
            className="btn btn-sm btn-basic-dark px-3 mr-2"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fa fa-chevron-up"
            />
          </button>
          <button
            className="btn btn-sm btn-basic-dark mr-2"
            type="button"
          >
            Copy Path
          </button>
          <a
            className="btn btn-sm btn-basic-dark"
            href="/#foo"
            onClick={[Function]}
          >
            View File
          </a>
        </div>
      </div>
      <div
        className="image-display "
      >
        
        
      </div>
    </div>
  </div>
</div>
`;
